src/ducks/balance/Balance.jsx

```diff
@@ -286,29 +290,76 @@ class Balance extends React.Component {
     this.props.router.push('/transactions')
   }
 
-  getBalanceHistory() {
-    const balanceHistories = getBalanceHistories(
-      historyData['io.cozy.bank.accounts'],
-      historyData['io.cozy.bank.operations']
-    )
+  getBalanceHistory(accounts, transactions) {
+    const balanceHistories = getBalanceHistories(accounts, transactions)
     const balanceHistory = sortBalanceHistoryByDate(balanceHistories.all)
 
     return balanceHistory
   }
 
-  getChartData() {
-    const history = this.getBalanceHistory()
+  getChartData(balanceHistory) {
     const WINDOW_SIZE = 15
 
-    const balancesSma = sma(history.map(h => h.y), WINDOW_SIZE, n => n)
+    const balancesSma = sma(balanceHistory.map(h => h.y), WINDOW_SIZE, n => n)
     const data = balancesSma.map((balance, i) => ({
-      ...history[i],
+      ...balanceHistory[i],
       y: balance
     }))
 
     return data
   }
 
+  renderHistory() {
+    const {
+      transactions: transactionsCollection,
+      accounts: accountsCollection
+    } = this.props
+
+    if (
+      isCollectionLoading(transactionsCollection) ||
+      isCollectionLoading(accountsCollection)
+    ) {
+      return 'Loading'
+    }
+
+    if (transactionsCollection.hasMore) {
+      transactionsCollection.fetchMore()
+      return 'Loading'
+    }
+
+    const history = this.getBalanceHistory(
+      accountsCollection.data,
+      transactionsCollection.data
+    )
+    const data = this.getChartData(history)
+    const nbTicks = uniq(
+      Object.keys(groupBy(data, i => formatDate(i.x, 'YYYY-MM')))
+    ).length
+    const intervalBetweenPoints = 57
+    const TICK_FORMAT = d3.timeFormat('%b')
+
+    return (
+      <History
+        accounts={accountsCollection.data}
+        chartProps={{
+          data,
+          nbTicks,
+          width: nbTicks * intervalBetweenPoints,
+          height: 103,
+          margin: {
+            top: 20,
+            bottom: 35,
+            left: 16,
+            right: 16
+          },
+          showAxis: true,
+          axisMargin: 10,
+          tickFormat: TICK_FORMAT
+        }}
+      />
+    )
+  }
+
```

Comment:
I'm thinking a dedicated component (linking the Balance and History) would be better than a method here. It would benefit from not being re-rendered if it is pure.

Reply:
Can you tell me more?

Reply:
We discussed with @drazik that we should then do a refacto to separate all that is graph in a separate duck ;)